maven-model-builder/src/main/java/org/apache/maven/model/building/ModelBuildingRequest.java

```diff
@@ -63,34 +63,35 @@
      * Denotes strict validation as recommended by the current Maven version.
      */
     int VALIDATION_LEVEL_STRICT = VALIDATION_LEVEL_MAVEN_3_0;
-    
+
     /**
-     * 
-     * @return the file model
+     * Gets the file model to build (with profile activatio). 
+     * If not set, model source will be used to load file model.
+     *
+     * @return The file model to build or {@code null} if not set.
      * @since 4.0.0
      */
     Model getFileModel();
-    
+
     /**
-     * 
+     * Set the file model with profile activation
+     *
      * @param fileModel
      * @return This request, never {@code null}.
      * @since 4.0.0
      */
     ModelBuildingRequest setFileModel( Model fileModel );
-
+    
     /**
-     * Gets the raw model to build. If not set, model source will be used to load raw model.
-     *
-     * @return The raw model to build or {@code null} if not set.
+     * @deprecated rawModel is never set, instead the fileModel is set 
```

Comment:
Why deprecate in a major version?

Reply:
I don't know if it is used by projects like polyglot and tiles. ModelBuildingRequest is used in an API.
Removing it now might imply that those projects can only be used with Maven3 or Maven4, not both. I don't think we should do that.